test/e2e-tests.sh

```diff
@@ -14,13 +14,34 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-source $(dirname $0)/e2e-common.sh
+# This script runs the end-to-end tests against net-gateway-api built from source.
+# It is started by prow for each PR. For convenience, it can also be executed manually.
+
+# If you already have a Knative cluster setup and kubectl pointing
+# to it, call this script with the --run-tests arguments and it will use
+# the cluster and run the tests.
+
+# Calling this script without arguments will create a new cluster in
+# project $PROJECT_ID, start knative in it, run the tests and delete the
+# cluster.
+
+# source "$(dirname $0)"/e2e-common.sh
+source $(dirname $0)/setup-and-deploy.sh
+
 
 # Script entry point.
 initialize "$@" --skip-istio-addon
 
+# Run the tests
+header "Running tests"
+
 failed=0
 
+function test_setup() {
+    deploy_contour
+    deploy_istio
+}
```

Comment:
This override has to be done before calling `initialize`

At this point the test is already running